### DISCUSSION TITLE
Recommendation for triangulation library

### DISCUSSION THREAD

**Discussion by solomancode** [yeslogic/allsorts#103] [Q&A] [answered]
Hi @wezm 

Thanks for answering my previous questions, I was able to implement glyph layout and placement correctly. I was looking for a triangulation library, I have tried [delaunay2d](https://crates.io/crates/delaunay2d) and [delaunator](https://crates.io/crates/delaunator) but I am unable to achieve a correct triangulation of a glyph. Below an Arabic glyph `Daal` as you can see the triangulation is not correct.

![Screenshot_20240226_160151](https://github.com/yeslogic/allsorts/assets/7840989/18f2f25c-9e22-4985-b990-3c22b863e055)

I was hoping to ask for your recommendation on the topic and if you can mention a library to use for glyph triangulation or a tutorial that discuss a technique to perform glyph triangulation correctly.

Thank you for your time,
Regards.

**Comment by wezm** [comment, marked as answer]
Sorry this is not something I have experience with. I assume triangulation is in service of GPU rendering? If so, both Raph Levien and Nical have done work on rendering 2D shapes on GPUs, perhaps that is useful?

- https://nical.github.io/posts/lyon-1-0.html
  - > [lyon_tessellation](https://docs.rs/lyon_tessellation/1.0.0/lyon_tessellation/)'s [fill tessellator](https://docs.rs/lyon_tessellation/1.0.0/lyon_tessellation/struct.FillTessellator.html) which produces a triangle mesh for any arbitrary vector path
- https://raphlinus.github.io/

**Comment by solomancode** [comment]
Thank you, Yes I need it for GPU rendering. Seems exactly what I'm looking for. Thanks again.
